### ISSUE TITLE
Avoid calling Plugin.Activate more than once

### ISSUE THREAD

**Issue by babu-shanmugam** [docker/docker#20815]
When docker is used with swarm, 'List' is called often, and
plugins.GetAll() is called every time during this. Since GetAll
uses loadWithRetry(), the plugin is initialized more than once.
This leaves a lot of connections to the plugin open and resulting on
Plugin.Activate called more than once

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

Output of `docker version`:

```
Docker version 1.11.0-dev, build 67a7793-unsupported
```

Output of `docker info`:

```
docker info from swarm:

Containers: 4
 Running: 4
 Paused: 0
 Stopped: 0
Images: 34
Server Version: swarm/1.1.2
Role: primary
Strategy: spread
Filters: health, port, dependency, affinity, constraint
Nodes: 1
 xxx: 192.168.1.100:2376
  └ Status: Healthy
  └ Containers: 4
  └ Reserved CPUs: 0 / 4
  └ Reserved Memory: 0 B / 8.065 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=3.13.0-79-generic, operatingsystem=Ubuntu 14.04.4 LTS, provider=generic, storagedriver=aufs
  └ Error: (none)
  └ UpdatedAt: 2016-03-01T13:34:23Z
Plugins: 
 Volume: 
 Network: 
Kernel Version: 3.13.0-79-generic
Operating System: linux
Architecture: amd64
CPUs: 4
Total Memory: 8.065 GiB
Name: xxx


docker info from engine:
Containers: 4
 Running: 4
 Paused: 0
 Stopped: 0
Images: 271
Server Version: 1.11.0-dev
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 327
 Dirperm1 Supported: false
Execution Driver: native-0.2
Logging Driver: json-file
Plugins: 
 Volume: local
 Network: null host bridge
Kernel Version: 3.13.0-79-generic
Operating System: Ubuntu 14.04.4 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.681 GiB
Name: xxx
ID: DDEQ:2VXR:IIO2:FNOB:V3AS:QIQ2:FLMA:3XGK:53TQ:O2AE:J3UM:XWDR
Debug mode (server): true
 File Descriptors: 45
 Goroutines: 85
 System Time: 2016-03-01T19:06:22.256812642+05:30
 EventsListeners: 1
 Init SHA1: 
 Init Path: 
 Docker Root Dir: /var/lib/docker
WARNING: No swap limit support
Labels:
 provider=generic
Cluster store: consul://192.168.1.100:8500
Cluster advertise: 192.168.1.100:2376


```

Provide additional environment details (AWS, VirtualBox, physical, etc.):
Physical

List the steps to reproduce the issue:
1. Run docker-machine with driver="generic" and setup a swarm master and swarm agent
2. Write a small VolumeDriver plugin as per the docker's VolumeDriver spec
3. For a regular interval, print the number of open connections to the plugin and check how many time Plugin.Activate is called

Describe the results you received:
For every time the docker engine receive "list volumes" (sometimes it gets this request automatically, I guess from the swarm process) request, a new connection will be open to the plugin and "Plugin.Activate" is called on that connection, after which 'VolumeDriver.List" is called.

Describe the results you expected:
Plugin.Activate should be called once as per the specification. Also, I expect not too many connections to be open by the docker engine to plugin

Provide additional info you think is important:

**Comment by srust** [comment]
I believe this should be closed because of #20686.

**Comment by cpuguy83** [comment]
Yep, thanks!
